File: euler/p10.cpp
#include <iostream>
#include <vector>
#include <math.h>
using namespace std;
vector<int> v{2, 3};
bool checkPrime(int num){
	if(num == 0 || num == 1) return false;
	int maxLoop = sqrt(num);
	for(int i: v){
		if(i > maxLoop) return true;
		if(num%i == 0) return false;
	}
	return true;
}
int main(void){
	long long result = 0;
	for(int i = 0; i < 2000000; i++){
		if(checkPrime(i)){
			v.push_back(i);
			result += i;
		}
	}
	cout << result;
}

File: euler/p10.py
from math import sqrt
primes = []
def checkPrime(num):
    maxLoop = sqrt(num)
    for i in primes:
        if i > maxLoop:
            return True
        if num%i == 0:
            return False
    return True

for i in xrange(2, 2000000):
    if(checkPrime(i)):
        primes.append(i)
print sum(primes)
